### ISSUE TITLE
Potential to utilize Parallel Programming in OC step

### ISSUE THREAD

**Issue by jiezhao1219** [VolpeUSDOT/FTOT-Public#15]
Hi @olivia-gillham-volpe @matthewpearlson 

Hope you are doing well!
We are now applying FTOT optimization into a more realistic supply chain (i.e. forest residuals-to-jet fuel supply chain), with incorporating the candidate generation process. However, due to the large size of this supply chain, it would take a minimum of 3 days only for oc step. In this case, is there any potential to adopt an effective approach in oc step to reduce the computational time, such as the Parallel Programming & Multiprocessing module in Python?
I appreciate your help!

Thanks, 
Jie Zhao (WSU)

**Comment by matthewpearlson** [comment]
Hi Jie. Thanks for reaching out about increasing the computation time for the OC step in complicated supply chains. I'm not certain off the top of my head if multi-processing modules can be used to speed up the candidate generation step of the optimizer. We have used the multiprocessing pools in the past to batch other steps, but I do not have any experience doing this in the optimizer. @olivia-gillham-volpe will know better than about this. I assigned this issue to her, but be aware she just returned from leave and may take a few days to respond. 

In the mean time, there are some other things we can do to try and improve the FTOT performance. For example, the minimum bounding geometry step in the C step still leaves a good portion of the national network in place for each of the modes. Perhaps removing all the links that are not in your area of interest could help reduce the amount of edges that need to be created for the network graph. Dane may be able to help you trim the network in ArcGIS. If he's not available, it may make sense for us to schedule some time to modify your network
